pipeline.py

```diff
@@ -72,31 +89,79 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
+        self.corpus_manager = corpus_manager
         pass
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        for article_item in self.corpus_manager.get_articles().values():
+            article_text = article_item.get_raw_text()
+            article_text = article_text.replace('-\n', '')
+            for symbol in ['\n', '\r']:
+                article_text = article_text.replace(symbol, ' ')
+            re_tokens = re.findall(r'[а-яё_]+-?[а-яё_]*', article_text, flags=re.IGNORECASE)
+            cleaned_text = ' '.join(re_tokens)
+            article_item.save_as(cleaned_text.lower(), 'cleaned')
+            self._process(cleaned_text)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        m_tokens_list = []
+        analyzed_text_mystem = Mystem().analyze(raw_text)
+        for analyzed_word in analyzed_text_mystem:
+            if analyzed_word['text'] == ' ':
+                continue
+            original_word = analyzed_word['text']
+            token = MorphologicalToken(original_word)
+            if 'analysis' not in analyzed_word.keys():
+                continue
+            if not analyzed_word['analysis']:
+                continue
+            if 'lex' not in analyzed_word['analysis'][0].keys():
+                continue
+            token.normalized_form = analyzed_word['analysis'][0].get('lex')
+            if 'gr' not in analyzed_word['analysis'][0].keys():
+                continue
+            token.tags_mystem = analyzed_word['analysis'][0].get('gr')
+            m_tokens_list.append(token)
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path_to_validate = Path(path_to_validate)
+    if not path_to_validate.is_dir():
+        raise NotADirectoryError
+    children_files = list(path_to_validate.glob('*raw.txt'))
+    children_files.extend(list(path_to_validate.glob('*.json')))
+    if not children_files:
+        raise InconsistentDatasetError
+    file_names = []
+    for files_path in children_files:
+        file_names.append(files_path.name)
+    if len(file_names) % 2 != 0:
+        raise InconsistentDatasetError
+    for i in range(1, int(len(list(children_files)) / 2) + 1):
+        if (f'{i}_raw.txt' not in file_names) or (f'{i}_meta.json' not in file_names):
+            raise FileNotFoundError
+    for file in children_files:
```

Comment:
what is the reason to open the file if it exists?

Reply:
To check if the file is empty or not? I believe we should have had a test for it. Do you want me to do it some other way?

Reply:
where did you find a requirement to open the file for emptiness? If this is not stated in docs, remove. Otherwise, please point to that place